### ISSUE TITLE
Add rule to check usage of variables defined in the same ENV statement

### ISSUE THREAD

**Issue by rodrigoaguilera** [hadolint/hadolint#509]
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

A warning is raised when a variable that is defined in the same ENV statement

### Actual behavior

No rule is triggered

### Steps to reproduce the behavior

**Dockerfile** 

```Dockerfile

ENV PACKAGE_NAME=${PACKAGE_NAME_ARG} \
  PACKAGE_BIN_DIR=/home/${PACKAGE_NAME}/vendor/bin

```
